server/src/controllers/christmas-tree.es6

```diff
@@ -528,30 +530,16 @@ christmasTree.updatePermitApplication = (req, res) => {
                 status: req.body.status
               })
               .then((permit)=>{
-                util.logControllerAction(req, 'christmasTree.updatePermitApplication', permit);
+                util.logControllerAction(req, 'christmasTree.updatePermitApplication#cancel', permit);
                 res.status(200).json(permit);
               });
           } else if (permit.status === 'Initiated' && req.body.status === 'Completed') {
-            const xmlData = paygov.getXmlToCompleteTransaction(permit.paygovToken);
-            postPayGov(xmlData).then(xmlResponse => {
-              parseXMLFromPayGov(res, xmlResponse, permit)
-                .then(paygovTrackingId => {
-                  return updatePermit(permit, {
-                    paygovTrackingId: paygovTrackingId,
-                    status: req.body.status
-                  });
-                })
-                .then(updatedPermit => {
-                  returnSavedPermit(res, permit);
-                  christmasTree.generateRulesAndEmail(updatedPermit);
-                })
-                .catch(error => {
-                  logger.error(error);
-                  throwError(error);
-                });
-            });
+            completePermitTransaction(permit, res, req);
```

Comment:
Would it make sense to know and take action on whether `completePermitTransaction ` completed successfully?

Reply:
Taking a stab at that... it actually would make a lot of sense.